### DISCUSSION TITLE
Budibase Cloud: There was a problem loading your grid. Cannot read properties of undefined (reading 'ta_users')

### DISCUSSION THREAD

**Discussion by stellarfinn** [Budibase/budibase#11713] [Help]
Today I logged in and I'm getting this error right off the bat. I haven't done anything in two weeks and had no issues. New update released and now this? 

<img width="1911" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/85e99d3d-7a7d-454a-a977-665934b6ba71">

This same error message shows for every table and ofcourse the frontend app doesn't load data. 

Why is this happening?

**Comment by stellarfinn** [comment]
Also, tried exporting the app and creating a new app. Same problem.

**Comment by shogunpurple** [comment]
@stellarfinn can you provide any kind of debugging information? Error in browser console, etc? Having trouble recreating this on any of our environments and hard to debug just from a screenshot

**Reply by stellarfinn** [reply]
@shogunpurple Thanks for your help!

<img width="843" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/07f17332-3148-422d-a8d4-476427fb8328">

<img width="1907" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/6f042edf-8b0b-4b26-9efd-859d2f25252a">

**Reply by stellarfinn** [reply]
Design view: 

<img width="1916" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/eaaaac59-2110-42c8-ab3a-96473a42f5b0">

**Comment by shogunpurple** [comment]
Fix released!

**Comment by sdsys-ch** [comment]
Just FYI:

Ran into a similar issue today:

`There was a problem loading your grid. Cannot read properties of undefined (reading 'primary')`

and the logs stated

```
bbapps                | {"level":"WARN","timestamp":"2024-02-28T09:36:08.896Z","service":"@budibase/server","err":{"type":"TypeError","message":"Cannot read properties of undefined (reading 'primary')","stack":"TypeError: Cannot read properties of undefined (reading 'primary')\n    at ExternalRequest.lookupRelations (/app/dist/index.js:188690:64)\n    at ExternalRequest.handleManyRelationships (/app/dist/index.js:188722:36)\n    at ExternalRequest.run (/app/dist/index.js:188883:22)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async Object.patch (/app/dist/index.js:188937:21)\n    at async patch3 (/app/dist/index.js:194564:42)\n    at async /app/node_modules/koa-compress/lib/index.js:38:5\n    at async errorHandling (/app/dist/index.js:156288:5)\n    at async /app/node_modules/koa-mount/index.js:52:26\n    at async userAgent (/app/node_modules/koa-useragent/dist/index.js:12:5)","expose":true,"statusCode":400,"status":400},"pid":29,"tenantId":"default","correlationId":"xxxxxxxx-xxxx-4b72-xxxx-xxxxxxxxxxxx","msg":""}
```

Turns out, all I had to do was _re-fetch the foreign-key tables_ (e.g. `jk_table1_table2_table2_table1`) from the PostgreSQL database that I had "un-fetched" previously via the Budibase UI, thinking that Budibase would have table-access anyhow and that it would make the `Data Sources` overview cleaner.

Adding the (Budibase-generated) tables back in the Budibase UI via `Fetch Tables` fixed it for me.
